src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
I was thinking that the `Planner` wrapper in `herbpy` would:

1. Check if the starting configuration is outside of `robot.GetDOFLimits()`. If it's not, then just continue as normal.
1. Otherwise, check if the starting configuration is outside of the hardware limits. If so, then there's some sort of error (perhaps the offsets are all wrong?) and the planner should definitely throw an error.
1. If we get here, then one or more of the starting configuration DOF values are outside the `DOFLimits` but inside the hardware limits. Compute new limits by extending the `DOFLimits` just enough to include the starting configuration, and set them on the robot temporarily by calling `SetDOFLimits()`. Then call the underlying planner as usual.  Make sure to reset the `DOFLimits` when the planner finishes (even in cases where the planner throws an exception) using a `finally` statement or a state saver.

The advantage here is that each planner (e.g. `or_ompl`) can continue to just use `GetDOFLimits()` and don't need to change. This logic could even just be implemented as a `prpy` meta-planner, which could be inserted into the `herbpy` planning pipeline.